tests/test_client.py

```diff
@@ -1931,11 +1930,12 @@ def test_update_camera_expired_token_success(self, mock_get, mock_put):
                   'state': 'IN', 'country': 'USA', 'longitude': '100.21323',
                   'latitude': '-44.9281', 'source': 'source', 'legacy_cameraID': None,
                   'm3u8_url': 'sample.com/m3u8_url', 'snapshot_url': 'sample.com/snapshot_url',
-                  'is_active_video': False, 'is_active_image': True, 'type': 'ip',
+                  'is_active_video': False, 'is_active_image': True, 'camera_type': 'ip',
                   'ip': '127.0.0.2', 'port': '8080', 'brand': 'Some Brand',
-                  'model': 'Some model', 'image_path': 'path/image', 'video_path': 'path/video'}
+                  'model': 'Some model', 'image_path': 'path/image', 'video_path': 'path/video',
+                  'cameraID': '5ae0ecbd336359291be74c12'}
 
```

Comment:
Can I format my update request as:

```
kwargs = {'cameraID': '5ae0ecbd336359291be74c12', 'camera_type': 'stream', 'm3u8_url': 'test_m3u8'}
resultID = client.write_camera(**kwargs)

```
In other words we should test PUT with just a change to the retreval model and none of the metadata. We should aslo test PUT with incorrect retreval method to insure updates work correctly. Such as:

```
kwargs = {'cameraID': '5ae0ecbd336359291be74c12', 'camera_type': 'non_ip', 'm3u8_url': 'test_m3u8'}
resultID = client.write_camera(**kwargs)
```

Reply:
for both case, yes you can and API will be able to detect whether the retrieval is compatible. If you did not specify the camera_type, API will check if the retrieval to be updated is compatible with the camera's current type. 
we could modify the testcases to look like that, although the logic is not taken cared by the python client. whatever returned by the API in each testcase is manually defined by the mock object.